tests/test_edge_cases.py

```diff
@@ -37,3 +39,22 @@ def g(x, y):
         return f(x, y)
 
     assert g(1, 2) == 3
+
+
+@dataclass(frozen=True)
+class Linear():
+    """Linear layer."""
+
+    W: object
+
+    def apply(self):
+        """Apply the layer."""
+        return self.W
+
+
+def test_get_dclass_fields():
+    from myia.utils import get_fields
+    lin = Linear(np.ones((2, 1)))
+    f = get_fields(lin)
+
+    assert eqtest(list(f)[0][1], np.ones((2, 1)))
```

Comment:
If you do the change I suggested in the other comment, I think this test is unnecessary.